augur/align.py

```diff
@@ -248,17 +255,32 @@ def read_reference(ref_fname):
                 "\n\tmake sure the file %s contains one sequence in genbank or fasta format"%ref_fname)
     return ref_seq
 
-def generate_alignment_cmd(method, nthreads, existing_aln_fname, seqs_to_align_fname, aln_fname, log_fname):
+def generate_alignment_cmd(method, nthreads, existing_aln_fname, seqs_to_align_fname, aln_fname, log_fname, alignment_args=None, override_default_args=False):
+    if method not in DEFAULT_ARGS:
+        raise AlignmentError('ERROR: alignment method %s not implemented'%method)
+    
+    if alignment_args is None:
+        alignment_args = DEFAULT_ARGS[method]
+    elif override_default_args:
+        alignment_args = alignment_args
+    else:
+        alignment_args = f"{DEFAULT_ARGS[method]} {alignment_args}"
+    
     if method=='mafft':
+        files_to_align = shquote(seqs_to_align_fname)
         if existing_aln_fname:
-            cmd = "mafft --add %s --keeplength --reorder --anysymbol --nomemsave --adjustdirection --thread %d %s 1> %s 2> %s"%(shquote(seqs_to_align_fname), nthreads, shquote(existing_aln_fname), shquote(aln_fname), shquote(log_fname))
-        else:
-            cmd = "mafft --reorder --anysymbol --nomemsave --adjustdirection --thread %d %s 1> %s 2> %s"%(nthreads, shquote(seqs_to_align_fname), shquote(aln_fname), shquote(log_fname))
+            # If there is an existing alignment, then seqs_to_align_fname becomes a parameter of --add
+            # and existing_aln_fname becomes the anonymous parameter
+            files_to_align = f"--add {shquote(seqs_to_align_fname)} {shquote(existing_aln_fname)}"
+            alignment_args = " ".join(["--keeplength", alignment_args])
+
+        cmd = f"mafft {alignment_args} --thread {nthreads} {files_to_align} 1> {shquote(aln_fname)} 2> {shquote(log_fname)}"
```

Comment:
It should be checked that `alignment_args` does not conflict with the hardcoded options (`--add`, `--keeplength`, `--thread`). See usage of [`augur.tree.check_conflicting_args`](https://github.com/nextstrain/augur/blob/eddc6157da1da1f6078a910b17af4f73633666b3/augur/tree.py#L59) as an example of how this is done in `augur tree`.

Actually, I think `augur.tree.check_conflicting_args` can be repurposed and used here by moving into `augur.utils` and replacing the "tree builder" references with something like "external program", but that's extra work. It's fine to copy the function as `augur.align.check_conflicting_args`.